components/bookingForm/bookingForm.tsx

```diff
@@ -22,246 +22,266 @@ import { Open_Sans } from "next/font/google";
-	const isShowStates = useMemo(() => handleStates(country), [country]);
+  //useMemo is call, whenever country value is changed
+  const isShowStates = useMemo(() => handleStates(country), [country]);
 
-	//Changing state of Validation Schema
-	const [schema, setSchema] = useState(() =>
-		ValidationSchema(isShowStates, false)
-	);
+  //Changing state of Validation Schema
+  const [schema, setSchema] = useState(() =>
+    ValidationSchema(isShowStates, false)
+  );
 
-	useEffect(() => {
-		// every time isShowState changes, recreate the schema and set it in the state
-		setSchema(ValidationSchema(isShowStates, false));
-	}, [isShowStates]);
+  const [invalidRecaptcha, setInvalidReptcha] = useState("");
 
-	const handleActiveInputLabel = (targetInput, value) => {
-		if (targetInput !== FORM_INPUT.States && !!value.trim()) {
-			setActiveInputLabel({ ...activeInputLabel, [targetInput]: true });
-		} else if (targetInput === FORM_INPUT.States && isShowStates) {
-			setActiveInputLabel({ ...activeInputLabel, [targetInput]: true });
-		} else {
-			setActiveInputLabel({ ...activeInputLabel, [targetInput]: false });
-		}
-	};
+  useEffect(() => {
+    // every time isShowState changes, recreate the schema and set it in the state
+    setSchema(ValidationSchema(isShowStates, false));
+  }, [isShowStates]);
 
-	const handleOnSubmit = async (values, actions) => {
-		const data = bookingFormSubmissionData(
-			values,
-			false,
-			contactReCaptcha,
-			sourceWebPageURL
-		);
-		actions.setSubmitting(false);
+  const handleActiveInputLabel = (targetInput, value) => {
+    if (targetInput !== FORM_INPUT.States && !!value.trim()) {
+      setActiveInputLabel({ ...activeInputLabel, [targetInput]: true });
+    } else if (targetInput === FORM_INPUT.States && isShowStates) {
+      setActiveInputLabel({ ...activeInputLabel, [targetInput]: true });
+    } else {
+      setActiveInputLabel({ ...activeInputLabel, [targetInput]: false });
+    }
+  };
 
-		await axios
-			.post("/api/create-lead", data)
-			.then(() => {
-				setContactSuccess(true);
-				setTimeout(function () {
-					setContactSuccess(false);
-					router.push("/thankyou/");
-				}, 1000);
-			})
-			.catch((err) => {
-				console.error(err);
-				return alert("Failed to create lead in CRM");
-			});
-	};
+  const handleOnSubmit = async (values, actions) => {
+    const data = bookingFormSubmissionData(
+      values,
+      false,
+      contactReCaptcha,
+      sourceWebPageURL
+    );
+    actions.setSubmitting(false);
 
-	const getCommonFieldProps = (fieldName: string) => ({
-		name: fieldName,
-		activeLabelClass: activeInputLabel[fieldName]
-			? ACTIVE_INPUT.ClassShow
-			: ACTIVE_INPUT.None,
-		handleChange: ({ name }, e) =>
-			handleActiveInputLabel(name, e.currentTarget.value),
-	});
+    await axios
+      .post("/api/validate-token", { contactReCaptcha })
+      .then(async (res) => {
+        if (res.data.success) {
+          setInvalidReptcha("");
```

Comment:
this form should call the create-lead endpoint
the endpoint should be doing the recaptcha validation internally as part of input validation